racket/collects/racket/port.rkt

```diff
@@ -164,6 +166,66 @@
 (define (call-with-input-bytes str proc)
   (with-input-from-x 'call-with-input-bytes 1 #t str proc))
 
+(define (combine-output-ports what . with)
+  (define port-list (cons what with))
+  (map (lambda (p)
+         (or (output-port? p)
+             (raise-argument-error 'combine-output-ports "output-port?" p))) port-list)
+  (define wrap-evt-with-evts
```

Comment:
It looks like this function is always called using `apply`, so it might make sense to just have it take a list as an argument.

Reply:
Thanks, these comments are very useful.
I've also been thinking about defining all the functions passed to `make-output-port` explicitly, instead of "on the spot". I feel like that would make the intent of the `combine-output-ports` function much clearer. What do you think?